### PR TITLE
docs(definition-list): add action popover and icon example - FE-3004

### PR DESCRIPTION
**Pull request by bulaj**
### Proposed behaviour
Available example of how to use ActionPopover and Icon within DefinitionList.
image

### Current behaviour
No available example of how to use ActionPopover and Icon within DefinitionList.

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [ ] <del>All themes are supported if required
- [ ] <del>Unit tests added or updated if required
- [ ] <del>Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [ ] <del>Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### Testing instructions
http://localhost:9001/?path=/docs/design-system-definition-list--action-popover-and-icon-support

### CODE REVIEW

**Review by sjohnson-sage — DISMISSED** [review]

**Review by dawidzarzycki — APPROVED** [review]

File: src/components/definition-list/definition-list.stories.mdx [hunk 1]
```diff
@@ -42,31 +48,75 @@ Definition List most of the time is in use with other components. Look at the ex
   </Story>
 </Preview>
 
+### Action popover and icon support
+
+<Preview>
+  <Story name="action popover and icon support">
+    <Dl>
+      <Dt>
+        <Box paddingTop="4px" display="inline-flex" alignItems="center">
```
Comment (sjohnson-sage) [inline src/components/definition-list/definition-list.stories.mdx:57]: We don't really need the `display` and `alignItems` props on this when it just has a string as a child

File: src/components/definition-list/definition-list.stories.mdx [hunk 2]
```diff
@@ -42,31 +48,75 @@ Definition List most of the time is in use with other components. Look at the ex
...
+        </Box>
+      </Dt>
+      <Dd>
+        <Box display="inline-flex" alignItems="center">
+          <Box mr={1}>Details example</Box>
+          <Icon type="tick" />
+        </Box>
+      </Dd>
+      <Dt>
+        <Box paddingTop="4px" display="inline-flex" alignItems="center">
```
Comment (sjohnson-sage) [inline src/components/definition-list/definition-list.stories.mdx:68]: Same as above

**Review by dawidzarzycki — APPROVED** [review]

File: src/components/tile/tile.stories.mdx [hunk 3]
```diff
@@ -379,6 +381,68 @@ In this example `dtTextAlign` has been provided the string 'left' and `ddTextAli
...
 </Preview>
 
+## With Definition List and ActionPopover and Icon support
+
+<Preview>
+  <Story name="with Definition List and ActionPopover and Icon support">
+    <Tile width={60}>
+      <Dl>
+      <Dt>
+        <Box paddingTop="4px" display="inline-flex" alignItems="center">
```
Comment (sjohnson-sage) [inline src/components/tile/tile.stories.mdx:391]: Same comment as above

File: src/components/tile/tile.stories.mdx [hunk 4]
```diff
@@ -379,6 +381,68 @@ In this example `dtTextAlign` has been provided the string 'left' and `ddTextAli
...
+        </Box>
+      </Dt>
+      <Dd>
+        <Box display="inline-flex" alignItems="center">
+          <Icon type="tick" mr={1}/>
+          <Box>Details example</Box>
+        </Box>
+      </Dd>
+      <Dt>
+        <Box paddingTop="4px" display="inline-flex" alignItems="center">
```
Comment (sjohnson-sage) [inline src/components/tile/tile.stories.mdx:402]: Same comment as above

File: src/components/tile/tile.stories.mdx [hunk 5]
```diff
@@ -379,6 +381,68 @@ In this example `dtTextAlign` has been provided the string 'left' and `ddTextAli
...
+        </Box>
+      </Dt>
+      <Dd>
+        <Box display="inline-flex" alignItems="center">
+          <Box mr={1}>Details example</Box>
+          <Icon type="tick" />
+        </Box>
+      </Dd>
+      <Dt>
+        <Box paddingTop="4px" display="inline-flex" alignItems="center">
```
Comment (sjohnson-sage) [inline src/components/tile/tile.stories.mdx:413]: Same comment as above

File: src/components/tile/tile.stories.mdx [hunk 6]
```diff
@@ -379,6 +381,68 @@ In this example `dtTextAlign` has been provided the string 'left' and `ddTextAli
...
+          <Icon mr={1} type="tick" />
+          <Box mr={2}>Details example</Box>
+          <ActionPopover rightAlignMenu>
+            <ActionPopoverItem>Option 1</ActionPopoverItem>
+            <ActionPopoverItem>Option 2</ActionPopoverItem>
+          </ActionPopover>
+        </Box>
+      </Dd>
+      <Dt>
+        <Box paddingTop="4px" display="inline-flex" alignItems="center">
```
Comment (sjohnson-sage) [inline src/components/tile/tile.stories.mdx:428]: Same comment as above

**Review by sjohnson-sage — APPROVED** [review]

**Review by dawidzarzycki — APPROVED** [review]

**Comment by ZhuoyuJin** [comment]
Tested in Safari, Chrome, Firefox, and Edge Chromium. No issues found. Testing passed.

**Review by ZhuoyuJin — APPROVED** [review]

**Pull request merged**